src/otx/v2/adapters/torch/mmengine/mmdet/modules/evaluation/evaluator.py

```diff
@@ -0,0 +1,469 @@
+                mious.append(cls_result["miou"])
+        mean_ap = np.array(aps).mean().item() if aps else 0.0
+        mean_miou = np.array(mious).mean().item() if mious else 0.0
+        metrics["mAP"] = mean_ap
+        metrics["mIoU"] = mean_miou
+
+        print_map_summary(mean_ap, eval_results, self.dataset_meta["classes"], None, logger=logger)
+
+        return metrics["mAP"], eval_results
+
+    def get_gt_masks(self, gts: list) -> list:
+        """Return masks information from gt."""
+        gt_masks = []
+        for idx in range(len(self.dataset_meta["classes"])):
+            gt_per_classes = []
+            for gt in gts:
+                gt_mask = gt["masks"][gt["labels"] == idx]
+                if len(gt_mask) == 0:
+                    gt_mask = []
+                gt_per_classes.append(gt_mask)
+            gt_masks.append(gt_per_classes)
+        return gt_masks
+
+    def compute_metrics(
+        self,
+        results: list,
+    ) -> OrderedDict[str, float]:
+        """Evaluate the dataset.
+
+        Args:
+            results (list): Testing results of the dataset.
+        """
+        logger: MMLogger = MMLogger.get_current_instance()
+        eval_results = OrderedDict()
+        gts, preds = zip(*results)
+        mean_aps = []
+        for iou_thr in self.iou_thrs:  # pylint: disable=redefined-argument-from-local
+            print_log(f'\n{"-" * 15}iou_thr: {iou_thr}{"-" * 15}')
+            if self.dataset_meta["domain"] == Domain.DETECTION:
+                mean_ap, _ = eval_map(
+                    preds,
+                    gts,
+                    scale_ranges=self.scale_ranges,
+                    iou_thr=iou_thr,
+                    dataset=self.dataset_meta["classes"],
+                    logger=logger,
+                    eval_mode=self.eval_mode,
+                    use_legacy_coordinate=True,
+                )
+            else:
+                mean_ap, _ = self.evaluate_mask(gts, preds, logger, iou_thr)
+            mean_aps.append(mean_ap)
+            eval_results[f"AP{int(iou_thr * 100):02d}"] = round(mean_ap, 3)
+        eval_results["mAP"] = sum(mean_aps) / len(mean_aps)
+        return eval_results
+
+    def process(self, data_batch: dict, data_samples: Sequence[dict]) -> None:  # noqa: ARG002
+        """Process one batch of data samples and predictions.
+
+        The processed results should be stored in ``self.results``, which will be used to
+        compute the metrics when all batches have been processed.
+
+        Args:
+            data_batch (dict): A batch of data from the dataloader.
+            data_samples (Sequence[dict]): A batch of data samples that
+                contain annotations and predictions.
+        """
+        for data_sample in data_samples:
+            gt = copy.deepcopy(data_sample)
+            gt_instances = gt["gt_instances"]
+            gt_ignore_instances = gt["ignored_instances"]
+            ann = {
```

Comment:
This might be opinionated but it reads/flows much better without abbreviations. So I suggest,
```suggestion
            annnotations = {
```